src/libsystemd-network/sd-ipv4acd.c

```diff
@@ -378,15 +379,33 @@ static int ipv4acd_on_packet(
 }
 
 int sd_ipv4acd_set_ifindex(sd_ipv4acd *acd, int ifindex) {
+        char ifname[IF_NAMESIZE + 1];
+
         assert_return(acd, -EINVAL);
         assert_return(ifindex > 0, -EINVAL);
         assert_return(acd->state == IPV4ACD_STATE_INIT, -EBUSY);
 
+        if (!format_ifname(ifindex, ifname))
+                return -ENODEV;
+
+        strcpy(acd->ifname, ifname);
         acd->ifindex = ifindex;
 
         return 0;
 }
 
+int sd_ipv4acd_get_ifindex(sd_ipv4acd *acd) {
+        assert_return(acd, -EINVAL);
```

Comment:
EBADF?

Maybe make this not assert_return but silent? See comments below...

Reply:
EBADF?? At least under libsystemd-network, we do not use `EBADF`...